File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/PaymentProduct771SpecificOutputMapper.cs
using PaymentProduct771SpecificOutputSdk = OnlinePayments.Sdk.Domain.PaymentProduct771SpecificOutput;
using PaymentProduct771SpecificOutputDto = Business.Domain.Payments.PaymentDetails.PaymentProduct771SpecificOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class PaymentProduct771SpecificOutputMapper
{
    public static PaymentProduct771SpecificOutputDto Map(PaymentProduct771SpecificOutputSdk? response)
    {
        return new()
        {
            MandateReference = response?.MandateReference
        };
    }
}

File: server/nodejs/src/infrastructure/mappers/payment-details/payment-product-771-specific-output-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { PaymentProduct771SpecificOutput } from '../../../business/domain/payments/payment-details/payment-product-771-specific-output';

type PaymentProduct771SpecificOutputSdk = Domain.PaymentProduct771SpecificOutput;

export const PaymentProduct771SpecificOutputMapper = {
    fromSdkResponse: (
        response?: PaymentProduct771SpecificOutputSdk | null,
    ): PaymentProduct771SpecificOutput | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            mandateReference: response?.mandateReference,
        };
    },
};
